src/CardDav/Backend.php

```diff
@@ -251,40 +233,40 @@ public function getLinkedData(string $uri): array
             'parameters' => $parameters ?? '',
             'data'       => (string)$response->getBody(),
         ];
+
         return $externalData;
     }
 
     /**
-     * Gets a clean vCard from the CardDAV server
+     * enrich the vcard with
+     * ->FULLNAME (equal to ->FN)
+     * ->LASTNAME, ->FIRSTNAME etc. extracted from ->N
+     * ->PHOTO with embedded data from linked sources (equal for KEY, LOGO or SOUND)
      *
-     * @param    string  $vcard_id   vCard id on the CardDAV server
-     * @return   stdClass            vCard (text/vcard)
+     * @param mixed $vcard
+     * @return mixed
      */
-    public function getVcard(string $vcard_id): stdClass
-    {
-        $vcard_id = str_replace($this->vcard_extension, '', $vcard_id) . $this->vcard_extension;
-        $response = $this->getCachedClient()->request('GET', $this->url . $vcard_id);
-
-        $body = (string)$response->getBody();
-        $vcard = $this->parseVcardFromContent($body);
-        $vcard = $this->enrichVcard($vcard);
-
-        return $vcard;
-    }
-
-    public function parseVcardFromContent(string $body): stdClass
+    public function enrichVcard($vcard)
     {
-        $parser = new Parser($body);
-        $vcard = $parser->getCardAtIndex(0);
-
-        return $vcard;
-    }
+        /** @var \stdClass $vcard */
+        if (isset($vcard->FN)) {                                // redundant for downward compatibility
+            $property = 'FULLNAME';
```

Comment:
I know this is stupid, but the only way to shup Travis CI up.
I tried
`$vcard->FULLNAME = (string)$vcard->FN;`
and
`$vcard->add('FULLNAME', (string)$vcard->FN);`
which are both correct - but Travis disagrees!

Reply:
Dennoch geht das so nicht. Wir verbasteln nicht den Code um Travis glücklich zu machen.

Reply:
built back to
`$vcard->FULLNAME = (string)$vcard->FN;`


as I said: Travis CI is a bastard - I really hate it :)